packages/sdk/src/utils/getTransactionSigner.ts

```diff
@@ -0,0 +1,21 @@
+import type { TransactionRequestLike } from 'fuels';
+import { hexlify, Address, InputType } from 'fuels';
+
+export function getTransactionSigner(transaction: TransactionRequestLike) {
+  const address = transaction.inputs?.map((i) => {
+    switch (i.type) {
+      case InputType.Message:
+        return i.recipient;
+      case InputType.Coin:
+        return i.owner;
+      default:
+        return undefined;
+    }
+  })[0];
+
+  if (!address) {
+    throw new Error('No possible signer found!');
+  }
+
+  return Address.fromB256(hexlify(address)).toString();
```

Comment:
this will always use [0] signer, is there a multi-signature level transaction that we need to cover? if yes can we create a follow up issue on that ?

Reply:
also this code could be inside `tx utils` , maybe adding some specific methods to deal with `TransactionRequest` / `TransactionRequestLike`

Reply:
I will not move to tx utils right now as it would make it difficult to use on the SDK creating a cycle dependency between SDK and Wallet once it's moved to `fuels-ts` we can replace this.

Open a issue for cover the multi-sig on the future https://github.com/FuelLabs/fuels-wallet/issues/522